### ISSUE TITLE
[Feature Request 1.16.4] Quest Reward Templates

### ISSUE THREAD

**Issue by MuteTiefling** [Direwolf20-MC/BuildingGadgets#563]
So I've been trying to work out how to give players an easy way to build certain structures, like Astral Sorcery or Mekanism multiblocks.

I thought that Building Gadget's was going to be perfect for this by storing them as templates and giving the templates as quest rewards. Sadly, this does not appear to be the case. It looks as though the templates point to some internal repository?

In any case, when our dev server was reset, all the templates I had made are now broken as they're no longer pointing to valid structures.  It seems that all of that information is stored in the World save.

Would it be possible to move that information somewhere else that could be included in a pack? This way the templates could be created and shared within the pack itself without the need to export and share them externally.

**Comment by MichaelHillcox** [comment]
Hi, yeah, I can see how this would be a problem, if you shipped the world specifically, and all of it's related files, you'd be able to have to working but for us to support this feature we'd need to support a way to allow direct referencing to a game file or something along those lines. Most mods do things like this to make it so other mods and the game itself don't have to try and parse a huge NBT tag inside of an item, and trust me, they're huge :P

**Comment by MuteTiefling** [comment]
Yeah, I can understand the desire to keep those huge strings off the item.

I'm thinking along the lines of ftb quests, actually, that keeps all of the quest info in a file structure in the config folder. Something like that would make it easily ship able.

You'd still keep your current ID system, just move where it's pointing to in the instance resources. 

Anyway, thanks for taking a look at this!

**Comment by hvdklauw** [comment]
Maybe we should have templates be loadable like datapacks, that we load the templates to the current world folder on world load and then they work like they currently do.

**Comment by MichaelHillcox** [comment]
Yeah, data packs have been floated around before. still not super sure if they would fit with a data pack, although, I haven't looked into how data packs work fully
